src/Settings.elm

```diff
@@ -0,0 +1,132 @@
+module Settings exposing (..)
+
+import Accessibility.Aria as A11y exposing (..)
+import Html exposing (Html)
+import Html.Attributes as Attrs exposing (..)
+import Html.Events exposing (onClick)
+import Set exposing (Set)
+import SharedTypes exposing (..)
+
+
+viewSettingsForm : UserGroup -> Set String -> (String -> msg) -> Html msg
+viewSettingsForm userGroup isInputVisible onInputShow =
+    let
+        -- Extract settings data from the user group
+        settings =
+            userGroup.settingsData
+
+        -- Determine if the settings are inherited or not
+        isInherited =
+            case settings.inheritedFrom of
+                Just "Null" ->
+                    False
+
+                _ ->
+                    True
+
+        -- Helper function to check if a value is present (Just) or not (Nothing)
+        hasValue : Maybe Int -> Bool
+        hasValue value =
+            case value of
+                Just _ ->
+                    True
+
+                Nothing ->
+                    False
```

Comment:
Tip: You can use [Maybe.Extra.isJust](https://package.elm-lang.org/packages/elm-community/maybe-extra/latest/Maybe-Extra#isJust), which renders this function a bit useless 😉 
```suggestion
        hasValue = Maybe.isJust
```

Reply:
Thank you for the tip 🙌